bellboy/actors/lcd.py

```diff
@@ -1,8 +1,128 @@
+import time
+from threading import Thread
+
+import gpiozero
+from gpiozero import DigitalOutputDevice
+from gpiozero.pins.mock import MockFactory
+from actors.generic import GenericActor
+from utils.messages import Response, LCDMsg, LCDReq, LCDResp, LCDEvent, LCDEventMsg,
 from bellboy.actors.generic import GenericActor
 
 
 # from bellboy.utils.messages import
 
 
+
 class LiquidCrystalActor(GenericActor):
+    """
+    Class for the LCD Display.
+
+    Contains a polling thread which can be run or stopped on message
+    request.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        # define private attributes
+        # the following are set on setup request
+
+        self._LCD = None
+
+    # --------------------------#
+    # STATE MODIFYING METHODS   #
+    # --------------------------#
+
+    def _setup_LCD(self, _sdaPin, sclPin):
+        """setup LCD paramaters."""
+        self._sdaPin = sdaPin
+        self._sclPin = sclPin
+
+        if self.TEST_MODE:
+            gpiozero.Device.pin_factory = MockFactory()
+            # TODO should this be "globally" set in the test suites...
+            # but then will it even be recognized in other domains...
+
+        self._LCD = LiquidCrystal(
+            sdaPin=self._sdaPin,
+            sclPin=self._sclPin
+
+        )
+
+        self.status = LCDResp.SET
+    
+    def displayLoop(self):
+        """running lcd thread. Displays text on screen"""
+
+        self.status = LCDResp.DISPLAY
+
+        while self.threadOn:
+            try:
+            self.log.info(
+                str.format("LCD DISPLAYS <<{}>>", )
```

Comment:
```
while self.threadOn:
    lcd.message(self._display_string)
```
 
no need for try

Reply:
That's true, the alignment here is kind of wonky too.